add arctan2 and curly brace validation

ChristopherChudzicki opened the pull request:
Edit: This PR addresses #133 and #138

jolyonb commented:
Want to update version numbers everywhere to 1.2.1? I think we should also include #138 in a version update.

Otherwise, this looks fine to me. Might suggest adding a couple of simple tests in the functions file where we test the inverse trig functions - arctan2(x, y) == arctan(y/x) type things.

ChristopherChudzicki commented:
Sure, I can implement #138 in this PR and add it to changelog. Either later tonight or tomorrow.

I just updated version in README.

I moved the arctan2 tests into their own file because of https://bugs.python.org/issue6835

jolyonb commented:
Looks good. Also remember to update `__version__` in `mitxgraders/version.py`

ChristopherChudzicki commented on mitxgraders/helpers/calc/expressions.py:
```diff
@@ -203,24 +207,28 @@ def raise_open_without_close(formula, stack):
+        if still_open:
+            message = "Invalid Input:\n"
+            for key in sorted_still_open:
+                message += ('{count} {name} were opened without being closed (highlighted below)\n'
```
I mangled the english a bit here (plural/singular not accounted for), but I don't think it matters and this is easier to maintain/extend.

jolyonb replied:
Was noticing this. You could add a "plural_name" to the dictionary, and put an if statement here on `count`.

jolyonb replied:
Given that it's a handful of lines and that this is unlikely to ever be touched again, might as well go for it, I think.

ChristopherChudzicki replied:
👍

jolyonb commented:
All looks good 👍
